### ISSUE TITLE
feat(dropdown): add translatable aria-labels to clrDropdownTrigger

### ISSUE THREAD

**Issue by bdryanovski** [vmware/clarity#3722]
## Summary

Implement translatable aria-labels to dropdowns.

* What is the change?

Add translatable aria-labels to clrDropdownTrigger button and change it based on the dropdown current state (open or close).

* Why should it go in Clarity?
Right now all dropdown doesn't have aria-label that is working out of the box and developer have to do it manually. Also, have to track the state of the dropdown by writing additional code.

* Does this change impact existing behaviors? If so how?
No

* If this change introduces a new behavior, is this behavior accessible?

## Use case
Right now adding aria-label to `clrDropdownTrigger` is done like this:

```html
<!-- Before open the dropdown -->
<button 
  class="nav-text" 
  clrDropdownTrigger 
  aria-label="Open dropdown"
>...</button>

<!-- After you open the dropdown you have to change it-->
<button 
  class="nav-text" 
  clrDropdownTrigger 
  aria-label="Close dropdown"
>...</button>
```
Quick workarround on that could be just to use the word `Toggle dropdown`. But this could not be ideal in all cases.

By adding `Input` we could:

* Simplify the implementation
* Automate it and write less code
* Get aria-label for all dropdowns out of the box

```html
<button 
  class="nav-text" 
  clrDropdownTrigger 
  clrAriaLabelDropdownOpen="open something something" 
  clrAriaLabelDropdownClose="close something something..."
>
```

```typescript
// dropdown-trigger.ts
host: {
  ...
 [attr.aria-label]: 'setAriaLabel',
  ...
}

get setAriaLabel() {
  return this.active ? this.clrAriaLabelDropdownOpen : this.clrAriaLabelDropdownClose;
}

...
@Input('clrAriaLabelDropdownOpen') clrAriaLabelDropdownOpen: string = commonStrings.keys.ariaLabelDropdownOpen;
@Input('clrAriaLabelDropdownClose') clrAriaLabelDropdownClose: string = commonStrings.keys.ariaLabelDropdownClose;
```

And the default for all dropdown will be
```typescript
// common-strings.default.ts
export const commonStringsDefault: ClrCommonStrings = {
  ...
  ariaLabelDropdownOpen: 'open dropdown',
  ariaLabelDropdownClose: 'close dropdown',
}
```

## Workarounds

* Add it by hand

**Comment by AlexMarcuBytex** [comment]
Hi @bdryanovski! Assign me so i can start to implement this workaround. Thanks!

**Comment by bdryanovski** [comment]
Sure, go on

**Comment by AlexMarcuBytex** [comment]
One more question. Shouldnt be aria-label set with "Open dropdown" status when this.active is false? 
Something link `return this.active ?  this.clrAriaLabelDropdownClose : this.clrAriaLabelDropdownOpen;`

**Comment by AlexMarcuBytex** [comment]
You want to be known the button will open a modal if its not active. otherwise, will close one.

**Comment by bdryanovski** [comment]
Yap, that is correct.

Like I said the above code is just for example so don't stick strictly to it.

**Comment by AlexMarcuBytex** [comment]
Alright. Thanks!

**Comment by gnomeontherun** [comment]
I just reviewed this issue and I disagree that this is something we should do. Sorry for the confusion, but this is overhead inside of Clarity to just set the `aria-label` on the button (which doesn't necessarily need this anyways, because buttons should have content). So the only thing that I see that could be actionable here is to write some documentation about how dropdown button triggers should have a meaningful label so people using screen readers can understand the button (general guidance for all buttons really), but if there is a case they use an icon button or need better context, then they can apply a custom `aria-label` on the app side. 

We only want to implement this special input pattern for adding `aria-label` customizations for interactive elements inside of the private DOM of a component, not when that DOM element is in the app context.

This should not be done by Clarity, at best it should be a documentation item which should just be covered by the accessibility documentation for buttons. Sorry for the run around here, but that means we should close this issue and close #3781.
